Lib/test/test_ctypes/test_c_simple_type_meta.py

```diff
@@ -69,7 +69,7 @@ def __new__(cls, name, bases, namespace):
                 if isinstance(self, p_meta):
                     return self
                 p = p_meta(f"POINTER({self.__name__})", (self, c_void_p), {})
-                ctypes._pointer_type_cache[self] = p
+                cls.__pointer_type__ = p
```

Comment:
Putting aside the dict-like operation, this `self` is an instance of a metaclass, meaning it is a class, so I think it should be written as follows:
```suggestion
                self.__pointer_type__ = p
```

Reply:
Good catch, thanks!